src/widgets.ts

```diff
@@ -83,6 +83,22 @@ export abstract class Widget<
     return dispatchWidgetPostMessageEvent
   }
 
+  /**
+   * Public method to communicate with iframe widget about navigation events
+   * Can be called when the host has a 'back' event happen.
+   * This will send a post message event to the iframe widget, which if it's
+   * listening for the 'mx/navigation' event, it can make its changes and then
+   * send its own post message event back to the SDK for the 'onNavigation' callback
+   */
+  navigateBack() {
```

Comment:
We want this interface/api to be the same or at least match the other sdks, so this is more of a note than anything else but let's keep the other sdks in mind as we do this.

Got any thoughts how we'll want to expose this functionality in the ios/android sdks?

Reply:
Yeah, that's a good question.

I figured this would have to be done on the widget level, since our post message definitions are all dynamically built, and we'd basically need a function that is accessible without a post message event firing to trigger the function. 

If we left the host to be in charge of the post message after capturing the 'back' event, they would send it directly to Connect, then we could skip this method call, since once Connect gets the event, it will return its own at which time the SDK would capture it and would fire its `onNavigation` callback function, which is the hosts hook into the event. But this feels bad, as I'd rather the SDK just handle the majority of the work, if not all. 

Happy to jump on a call and discuss some other options. Maybe I misinterpreted your original issue when creating this.

Reply:
I'm talking about the API across all of the SDKs. There will have to be some differences on how a client interacts with our widgets/SDK depending on the environment they're in, but we want consistency across all SDK with the functionality we expose, how it's accessed, how it works, the language/words/names we use, etc. For example, in the Web SDK we'll have something like:

```js
const widget = new ConnectWidget({ /* ... */ })

GlobalNavigation.onBackButtonClick(async (event) => {
  const didGoBack = await widget.goBack()
  if (!didGoBack) {
    GlobalNavigation.goBack()
  }
})
```

React Native, it could be:

```jsx
<ConnectWidget
  navigation={(goBack) => {
    GlobalNavigation.onBackButtonClick((event) => {
      goBack().then(didGoBack => {
        if (!didGoBack) {
          GlobalNavigation.goBack()
        }
      })
    })
  }}
/>
```


iOS could be:

```swift
self.widget = ConnectWidget({ /* ... */ })
self.button = BlockBarButtonItem(title: "Back", actionHandler: { () in
  let didGoBack = await self.widget.goBack()
  if (!didGoBack) {
    self.navigationController.popViewControllerAnimated(true)
  }
})
```

Ultimately what I'm saying is let's make sure the names and overall functionality we use here will be able to be carried over to the other SDKs.

Reply:
I use `goBack` in my example because that's what I originally though of calling it but I could go with `navigateBack`. Another thing: we should also update our examples in each SDK repo to use the back functionality once it's implemented.